### DISCUSSION TITLE
[Community] Help us test Superset's experience with different databases

### DISCUSSION THREAD

**Discussion by srinify** [apache/superset#19256] [Ideas]
Through SQLAlchemy and Python DB-API 2.0, Superset can query potentially hundreds of SQL-speaking databases. The community documentation maintains an incomplete list here: https://superset.apache.org/docs/databases/dockeradddrivers

Testing all of these databases with each minor or major release is a lot of work that can't be carried out by a single team or organization! In addition, navigating the nuances of each database is best done by people actively using the database for their own needs. This is an ideal candidate for a community effort 🌻 

**As a first step, I'm looking to build a list of Database Champions who want to help with documenting, answering questions about the specific database(s), or even testing / ensuring that Superset's integration with the database(s) is maintained!**.

Some common databases that I know at least a few community members use in conjunction with Superset:

- MySQL
- Postgres
- PrestoDB
- Trino
- Hive
- Druid
- Pinot
- Clickhouse
- BigQuery
- Redshift

If you want to help out with testing or be available as a resource for others in the community, **please comment in this thread with the databases you want to champion for**. We'll also be using the #contributing channel in the [Superset Community Slack](https://join.slack.com/t/apache-superset/shared_invite/zt-l5f5e0av-fyYu8tlfdqbMdz_sPLwUqQ) to discuss improving support for database connections. 

Thank you to Sophie @betodealmeida and etc. for helping kickstart this initiative!

**Comment by garden-of-delete** [comment]
I can champion Druid to start. I can further pick up Postgres, Redshift, and PrestoDB if needed down the line as well, but would prefer to distribute the knowledge more widely if possible.

**Comment by aakashnand** [comment]
@srinify as discussed in this slack thread, I am still stuck at getting started trino with the superset. I will try it again and once I get access I would like to help with trino db.
https://apache-superset.slack.com/archives/C015WAZL0KH/p1614923108050900

**Comment by zach-data** [comment]
I’m happy to own Redshift.

**Comment by TheRum** [comment]
Would love to help testing Druid !

**Comment by inviscid** [comment]
Although very similar to Postgres, Greenplum is a database we use a lot.

**Reply by pankajsoni22** [reply]
I am not able to connect with greenplum, can someone help me connecting that.

**Comment by rvazquezglez** [comment]
I've been testing creating some visualizations for a Trino instance connected to MongoDB. Happy to help.

**Reply by Deninc** [reply]
Hi @rvazquezglez, how is your experience so far. I'm considering between Trino, Apache Drill vs MongoDB BI Connector.

**Reply by FrankHuy** [reply]
Hi @Deninc @rvazquezglez , how is it going? I'm finding a solation for adding MongoDB connections to Superset, too.

**Reply by ziggekatten** [reply]
We replaced Drill with Trino, and doing so got our MongoDB connections to just work. Drill is falling behind it seems

**Comment by rkpotdar** [comment]
@srinify Been testing Clickhouse database connection and happy to help more.
At the moment Clickhouse connection in Superset doesn't allow CSV uploads or create table operation as the Engine details are needed in create query and Superset doesn't have option to add these details during CSV upload.

**Comment by rvazquezglez** [comment]
If you are using just one db instance Mongo DB BI connector is the best
integration.

However if you have several instances (like in a microservices
architecture, each service with its own db) you'll need trino or drill.

I haven't used drill in deep because it didn't support Mongo DBRefs at the
time I tried (and it was one of my main requirements). Also I felt drill
setup harder than trino.

Trino has worked fine for me. The only drawback I'd say is that it doesn't
convert types automatically and that sometimes clashes with superset custom
inputs for visualizations (e. g. Superset generating dates groupings with
dates as strings and trino not accepting them because it requires date
types rather than the date strings), but I've worked around that with trino
castings around the `{{ from_dttm }}` and `{{ to_dttm}}` Jinja variables.
